problemtools/verifyproblem.py

```diff
@@ -182,12 +185,15 @@ def _check_symlinks(self):
             return False
         return True
 
-    def run_submission(self, sub, args, timelim_low, timelim_high):
+    def run_submission(self, sub, args, timelim_low, timelim_high, subinfo):
```

Comment:
I don't like introducing this dict of "whatever extra data we realize we want to pass around".  In this case it is not even really the case of info that is being passed to `run_submission`, but rather a mechanism for `run_submission` to send additional result data back.

I would prefer adding a "failed_samples" field to the SubmissionResult class and passing it back upwards that way.  (The SubmissionResult class is already suffering from being a bit of a "let's add everything that occurs to us here" mess, but at least then that messiness remains contained to that class and we're not adding another mechanism with the same issues.)

Reply:
Makes sense. I'll look into this and see if it's doable.